09_2024_scikit_learn_priorities.md

```diff
@@ -0,0 +1,260 @@
+
+1. Support for `polars` dataframes in `skrub` estimators.
+1. Adding interactive reports of dataframes through a `TableReport` that provides
+   information to carry out Exploratory Data Analysis (EDA).
+1. Easing the creation of predictive models by removing boilerplate code using a factory
+   function called `tabular_learner`.
+1. Improving the landing page of the `skrub` website to make it more user-friendly and
+   informative.
+
+### Python in the browser (WebAssembly)
+
+We make sure that `scikit-learn` is compatible with the WebAssembly stack:
+[`pyodide`](https://github.com/pyodide/pyodide),
+[`jupyterlite`](https://github.com/jupyterlite/jupyterlite). We reported potential
+issues upstream and helped run the SciPy test suite for the pyodide project.
+
+### [`hazardous`](https://github.com/soda-inria/hazardous)
+
+The current focus for this project is to define the scope of the library such that it
+does not overlap with existing tools. While the code is developed in parallel with a
+research project, we are working on improving tests and documentation to make the
+library more robust and ready for a first release.
+
+## Focus Areas for the Next 6 Months
+
+As we look ahead to the next six months, we have identified several key areas where we
+will concentrate our efforts to further enhance `scikit-learn` and its ecosystem. These
+focus areas align with our general objectives and aim to address current challenges
+and opportunities in the machine learning landscape.
+
+### [`scikit-learn`](https://github.com/scikit-learn/scikit-learn)
+
+We start by focusing on the future work for `scikit-learn`.
+
+1. Statistical correctness: *ogrisel, snath-xoc, antoinebaker, jeremiedbb*
+   - We will continue our maintenance work to ensure that sample weights are
+      correctly handled in estimators.
+   - We will improve the documentation to better explain which cross-validation
+      strategies should be used in line with the end-user's goal.
+   - We will work on improving the solvers of linear models.
+1. New visualizations: *lucyleeow, glemaitre, jeremiedbb*
+   - We will develop some additional displays to help with model inspection.
+   - We will work on adapting the existing displays to visualize the outputs of
+     cross-validation results.
+   - We will search for ways to improve the HTML representation of estimators.
+1. Callbacks API: *jeremiedbb, adrinjalali, glemaitre, ogrisel*
+   - We will work on merging the infrastructure for callbacks together with a callback
+     to provide progress bars.
+   - Subsequently, we will work on additional callbacks.
+1. Improve histogram gradient-boosting: *GaelVaroquaux, glemaitre, lesteve, ogrisel*
+   - We will benchmark the `HistGradientBoosting` estimators to understand the room
+     for improvement of these estimators to match the performance of LightGBM and
+     XGBoost.
+   - We will assess if we should merge both `GradientBoosting` and
```

Comment:
Isn't this a community decision?

Reply:
Of course. `Assess` here means that someone will report the pros/cons, potential difficulties, by looking in details if we want to go this road. The final call will be based on the consensus.

I will rephrase it because the `we should merge` does not reflect the consensus base decision.